src/plugins/chart_expressions/expression_xy/public/helpers/axes_configuration.ts

```diff
@@ -131,11 +131,26 @@ export function groupAxesByType(
 
     let axisGroupId = LEFT_GLOBAL_AXIS_ID;
 
-    if (series[LEFT_GLOBAL_AXIS_ID].length === 0 || leftAxisGroupId) {
+    const rightSeriesCount = rightSeriesKeys.reduce((acc, key) => {
```

Comment:
I don't really understand from the code what `rightSeriesKeys` and `leftSeriesKeys` actually are - it seems like they are a mixture of explicit axis ids and the string `auto`? (pretty sure this is also causing the bug)